tuf/download.py

```diff
@@ -192,33 +192,33 @@ def _download_file(url, required_length, fetcher, STRICT_REQUIRED_LENGTH=True):
   # This is the temporary file that we will return to contain the contents of
   # the downloaded file.
   temp_file = tempfile.TemporaryFile()
-  start_time = timeit.default_timer()
 
   average_download_speed = 0
   number_of_bytes_received = 0
 
   try:
-
-    for chunk in fetcher.fetch(url, required_length):
+    chunks = fetcher.fetch(url, required_length)
+    start_time = timeit.default_timer()
+    for chunk in chunks:
 
       stop_time = timeit.default_timer()
-      seconds_spent_receiving = stop_time - start_time
+      temp_file.write(chunk)
```

Comment:
It seems I  am missing context, but why did you decide to move the write chunk to temp file before exiting the fetcher loop on error?

Reply:
I believe it's because the old code can lead to an exception even if all bytes have already been downloaded (which seems a bit useless)

Reply:
In both cases writing the chunk to the temp_file is in the for loop. I moved it to be before the average download speed calculation and mostly before the `break` statement on line 216 since it was resulting in misleading errors on low average download speed as Jussi has explained.

Reply:
@MVrachev ah sorry, misread your question, so yeah ... what Jussi said :grin: